File: java/app/src/main/java/tw/com/ecpay/paymentgatewaykit/example/util/AlertUtil.java
package tw.com.ecpay.paymentgatewaykit.example.util;

import android.content.Context;
import android.content.DialogInterface;

import androidx.appcompat.app.AlertDialog;

public class AlertUtil {

    private AlertUtil() {
    }

    public static void showAlertDialog(Context context,
                                       String title,
                                       String message,
                                       DialogInterface.OnClickListener listenerLeftButton,
                                       String leftButtonName,
                                       DialogInterface.OnClickListener listenerRightButton,
                                       String rightButtonName) {
        AlertDialog.Builder dialog = new AlertDialog.Builder(context);
        dialog.setTitle(title);
        dialog.setMessage(message);
        dialog.setNeutralButton(leftButtonName, listenerLeftButton);
        dialog.setNegativeButton(rightButtonName, listenerRightButton);
        dialog.show();
    }

    public static void showAlertDialog(Context context,
                                       String title,
                                       String message,
                                       DialogInterface.OnClickListener listenerRightButton,
                                       String rightButtonName) {
        AlertDialog.Builder dialog = new AlertDialog.Builder(context);
        dialog.setTitle(title);
        dialog.setMessage(message);
        dialog.setNegativeButton(rightButtonName, listenerRightButton);
        dialog.show();
    }
}

File: kotlin/app/src/main/kotlin/tw/com/ecpay/paymentgatewaykit/example/util/AlertUtil.kt
package tw.com.ecpay.paymentgatewaykit.example.util

import android.app.AlertDialog
import android.content.Context
import android.content.DialogInterface

class AlertUtil {

    companion object {

        fun showAlertDialog(
            context: Context?,
            title: String?,
            message: String?,
            listenerRightButton: DialogInterface.OnClickListener?,
            rightButtonName: String?
        ) {
            val dialog: AlertDialog.Builder = AlertDialog.Builder(context)
            dialog.setTitle(title)
            dialog.setMessage(message)
            dialog.setNegativeButton(rightButtonName, listenerRightButton)
            dialog.show()
        }
    }
}